### ISSUE TITLE
Build error on Ubuntu 16.04

### ISSUE THREAD

**Issue by daichen** [google/filament#2889]
**Describe the bug**
Build failed on Ubuntu 16.04
cmake 3.18
clang 8

**To Reproduce**
Command : ./build.sh -c debug

Cleaning build directories...
Building debug in out/cmake-debug...
-- The C compiler identification is Clang 8.0.0
-- The CXX compiler identification is Clang 8.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- The ASM compiler identification is Clang
-- Found assembler: /usr/bin/cc
-- DFG LUT size set to 128x128
-- Found Python3: /usr/bin/python3.5 (found version "3.5.2") found components: Interpreter 
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.5.2", minimum required is "3") 
-- Found Java: /usr/lib/jvm/java-8-openjdk-amd64/bin/java (found version "1.8.0_252") 
-- Found JNI: /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/libjawt.so  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mi/work/code/filament/out/cmake-debug
[14/1275] Building CXX object third_party/libgtest/tnt/CMakeFiles/gtest.dir/__/googletest/src/gtest-all.cc.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../third_party/libgtest/tnt/../googletest/include -I../../third_party/libgtest/tnt/../googletest -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -pthread -MD -MT third_party/libgtest/tnt/CMakeFiles/gtest.dir/__/googletest/src/gtest-all.cc.o -MF third_party/libgtest/tnt/CMakeFiles/gtest.dir/__/googletest/src/gtest-all.cc.o.d -o third_party/libgtest/tnt/CMakeFiles/gtest.dir/__/googletest/src/gtest-all.cc.o -c ../../third_party/libgtest/googletest/src/gtest-all.cc
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/camutils/CMakeFiles/camutils.dir/src/Bookmark.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/camutils/include -I../../libs/math/include -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -Wno-deprecated-register -MD -MT libs/camutils/CMakeFiles/camutils.dir/src/Bookmark.cpp.o -MF libs/camutils/CMakeFiles/camutils.dir/src/Bookmark.cpp.o.d -o libs/camutils/CMakeFiles/camutils.dir/src/Bookmark.cpp.o -c ../../libs/camutils/src/Bookmark.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/camutils/CMakeFiles/camutils.dir/src/Manipulator.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/camutils/include -I../../libs/math/include -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -Wno-deprecated-register -MD -MT libs/camutils/CMakeFiles/camutils.dir/src/Manipulator.cpp.o -MF libs/camutils/CMakeFiles/camutils.dir/src/Manipulator.cpp.o.d -o libs/camutils/CMakeFiles/camutils.dir/src/Manipulator.cpp.o -c ../../libs/camutils/src/Manipulator.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/camutils/CMakeFiles/test_camutils.dir/tests/test_camutils.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/camutils/include -I../../libs/math/include -I../../third_party/libgtest/tnt/../googletest/include -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -pthread -MD -MT libs/camutils/CMakeFiles/test_camutils.dir/tests/test_camutils.cpp.o -MF libs/camutils/CMakeFiles/test_camutils.dir/tests/test_camutils.cpp.o.d -o libs/camutils/CMakeFiles/test_camutils.dir/tests/test_camutils.cpp.o -c ../../libs/camutils/tests/test_camutils.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filabridge/CMakeFiles/filabridge.dir/src/SamplerBindingMap.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector -MD -MT libs/filabridge/CMakeFiles/filabridge.dir/src/SamplerBindingMap.cpp.o -MF libs/filabridge/CMakeFiles/filabridge.dir/src/SamplerBindingMap.cpp.o.d -o libs/filabridge/CMakeFiles/filabridge.dir/src/SamplerBindingMap.cpp.o -c ../../libs/filabridge/src/SamplerBindingMap.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filabridge/CMakeFiles/filabridge.dir/src/UibGenerator.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector -MD -MT libs/filabridge/CMakeFiles/filabridge.dir/src/UibGenerator.cpp.o -MF libs/filabridge/CMakeFiles/filabridge.dir/src/UibGenerator.cpp.o.d -o libs/filabridge/CMakeFiles/filabridge.dir/src/UibGenerator.cpp.o -c ../../libs/filabridge/src/UibGenerator.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filabridge/CMakeFiles/filabridge.dir/src/UniformInterfaceBlock.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector -MD -MT libs/filabridge/CMakeFiles/filabridge.dir/src/UniformInterfaceBlock.cpp.o -MF libs/filabridge/CMakeFiles/filabridge.dir/src/UniformInterfaceBlock.cpp.o.d -o libs/filabridge/CMakeFiles/filabridge.dir/src/UniformInterfaceBlock.cpp.o -c ../../libs/filabridge/src/UniformInterfaceBlock.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filabridge/CMakeFiles/filabridge.dir/src/SamplerInterfaceBlock.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector -MD -MT libs/filabridge/CMakeFiles/filabridge.dir/src/SamplerInterfaceBlock.cpp.o -MF libs/filabridge/CMakeFiles/filabridge.dir/src/SamplerInterfaceBlock.cpp.o.d -o libs/filabridge/CMakeFiles/filabridge.dir/src/SamplerInterfaceBlock.cpp.o -c ../../libs/filabridge/src/SamplerInterfaceBlock.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filabridge/CMakeFiles/filabridge.dir/src/SibGenerator.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector -MD -MT libs/filabridge/CMakeFiles/filabridge.dir/src/SibGenerator.cpp.o -MF libs/filabridge/CMakeFiles/filabridge.dir/src/SibGenerator.cpp.o.d -o libs/filabridge/CMakeFiles/filabridge.dir/src/SibGenerator.cpp.o -c ../../libs/filabridge/src/SibGenerator.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filaflat/CMakeFiles/filaflat.dir/src/DictionaryReader.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filaflat/include -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -I../../third_party/smol-v/tnt/../source -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -fvisibility=hidden -MD -MT libs/filaflat/CMakeFiles/filaflat.dir/src/DictionaryReader.cpp.o -MF libs/filaflat/CMakeFiles/filaflat.dir/src/DictionaryReader.cpp.o.d -o libs/filaflat/CMakeFiles/filaflat.dir/src/DictionaryReader.cpp.o -c ../../libs/filaflat/src/DictionaryReader.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filaflat/CMakeFiles/filaflat.dir/src/ChunkContainer.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filaflat/include -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -I../../third_party/smol-v/tnt/../source -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -fvisibility=hidden -MD -MT libs/filaflat/CMakeFiles/filaflat.dir/src/ChunkContainer.cpp.o -MF libs/filaflat/CMakeFiles/filaflat.dir/src/ChunkContainer.cpp.o.d -o libs/filaflat/CMakeFiles/filaflat.dir/src/ChunkContainer.cpp.o -c ../../libs/filaflat/src/ChunkContainer.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filaflat/CMakeFiles/filaflat.dir/src/MaterialChunk.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filaflat/include -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -I../../third_party/smol-v/tnt/../source -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -fvisibility=hidden -MD -MT libs/filaflat/CMakeFiles/filaflat.dir/src/MaterialChunk.cpp.o -MF libs/filaflat/CMakeFiles/filaflat.dir/src/MaterialChunk.cpp.o.d -o libs/filaflat/CMakeFiles/filaflat.dir/src/MaterialChunk.cpp.o -c ../../libs/filaflat/src/MaterialChunk.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filaflat/CMakeFiles/filaflat.dir/src/ShaderBuilder.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filaflat/include -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -I../../third_party/smol-v/tnt/../source -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -fvisibility=hidden -MD -MT libs/filaflat/CMakeFiles/filaflat.dir/src/ShaderBuilder.cpp.o -MF libs/filaflat/CMakeFiles/filaflat.dir/src/ShaderBuilder.cpp.o.d -o libs/filaflat/CMakeFiles/filaflat.dir/src/ShaderBuilder.cpp.o -c ../../libs/filaflat/src/ShaderBuilder.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
[14/1275] Building CXX object libs/filaflat/CMakeFiles/filaflat.dir/src/Unflattener.cpp.o
FAILED: /home/mi/work/code/filament/out/cmake-debug/launch-cxx /usr/bin/c++ -DFILAMENT_DRIVER_SUPPORTS_VULKAN -I../../libs/filaflat/include -I../../libs/filabridge/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/math/include -I../../filament/backend/include -I../../third_party/smol-v/tnt/../source -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -fvisibility=hidden -MD -MT libs/filaflat/CMakeFiles/filaflat.dir/src/Unflattener.cpp.o -MF libs/filaflat/CMakeFiles/filaflat.dir/src/Unflattener.cpp.o.d -o libs/filaflat/CMakeFiles/filaflat.dir/src/Unflattener.cpp.o -c ../../libs/filaflat/src/Unflattener.cpp
/home/mi/work/code/filament/out/cmake-debug/launch-cxx: 3: /home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"
ninja: build stopped: subcommand failed.

**Comment by romainguy** [comment]
Looks like your ccache is too old maybe?

```/home/mi/work/code/filament/out/cmake-debug/launch-cxx: [[: not found
ccache: error: SLOPPINESS: unknown sloppiness: "locale"```

You can try running without it.

**Comment by daichen** [comment]
after remove ccache
I meet another error, I' using clang-8  libc++-8-dev libc++abi-8-dev

./build.sh -p android release
Building release in out/cmake-release...
-- The C compiler identification is Clang 8.0.0
-- The CXX compiler identification is Clang 8.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- The ASM compiler identification is Clang
-- Found assembler: /usr/bin/cc
-- DFG LUT size set to 128x128
-- Found Python3: /usr/bin/python3.5 (found version "3.5.2") found components: Interpreter 
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.5.2", minimum required is "3") 
-- Found Java: /usr/lib/jvm/java-8-openjdk-amd64/bin/java (found version "1.8.0_252") 
-- Found JNI: /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/libjawt.so  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mi/work/code/filament/out/cmake-release
[516/672] Linking CXX executable tools/resgen/resgen
FAILED: : && /usr/bin/c++ -std=c++14 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -stdlib=libc++ -fPIC -fcolor-diagnostics -O3 -DNDEBUG -fomit-frame-pointer -ffunction-sections -fdata-sections -Wl,--gc-sections tools/resgen/CMakeFiles/resgen.dir/src/main.cpp.o -o tools/resgen/resgen  -static-libgcc -static-libstdc++  -Wl,-Bstatic  -lc++  -lc++abi  libs/utils/libutils.a  third_party/getopt/libgetopt.a  -Wl,-Bdynamic  -pthread  -static-libgcc -static-libstdc++  -Wl,-Bstatic  -lc++  -lc++abi  -Wl,-Bdynamic && :
clang: error: unable to execute command: Segmentation fault (core dumped)
clang: error: linker command failed due to signal (use -v to see invocation)
[516/672] Building CXX object third_party/spirv-cross/tnt/CMakeFiles/spirv-cross-msl.dir/__/spirv_msl.cpp.o
ninja: build stopped: subcommand failed.

**Comment by romainguy** [comment]
Apparently your linker crashed. I don't think there's anything we can do about this, it looks like a linker bug. Best bet is to update your toolchain?

**Comment by ab503044120** [comment]
I fix it by update ccache version to 4.1
